src/solvers/refinement/string_constraint.h

```diff
@@ -204,6 +208,21 @@ class string_not_contains_constraintt: public exprt
   }
 };
 
+inline static std::string from_expr(
+  const namespacet &ns,
+  const irep_idt &identifier,
+  const string_not_contains_constraintt &expr)
```

Comment:
After we talked in real life, we decided that this is the best choice, where preferably in the future `from_expr` is refactored and possibly replaced.